haystack/retriever/dense.py

```diff
@@ -105,158 +185,97 @@ def embed_passages(self, docs: List[Document]) -> List[np.array]:
-            # remove [SEP] token from untitled passages in batch
-            if self.embed_title and self.remove_sep_tok_from_untitled_passages and ctx_title:
-                ctx_ids_batch, ctx_attn_mask = self._remove_sep_tok_from_untitled_passages(ctx_title,
-                                                                                           ctx_ids_batch,
-                                                                                           ctx_attn_mask)
-
-            with torch.no_grad():
-                out = model(input_ids=ctx_ids_batch, attention_mask=ctx_attn_mask, token_type_ids=ctx_seg_batch)
-                # TODO revert back to when updating transformers
-                # out = out.pooler_output
-                out = out[0]
-            out = out.cpu()
-
-            total += ctx_ids_batch.size()[0]
-
-            results.extend([
-                (out[i].view(-1).numpy())
-                for i in range(out.size(0))
-            ])
-
-            if total % 10 == 0:
-                logger.info(f'Embedded {total} / {n} texts')
-
-        return results
+        self.embed_title = embed_title
+        self.processor = TextSimilarityProcessor(tokenizer=self.query_tokenizer,
+                                                 passage_tokenizer=self.passage_tokenizer,
+                                                 max_seq_len_passage=self.max_seq_len_passage,
+                                                 max_seq_len_query=self.max_seq_len_query,
+                                                 label_list=label_list,
+                                                 metric=metric,
+                                                 data_dir=data_dir,
+                                                 train_filename=train_filename,
+                                                 dev_filename=dev_filename,
+                                                 test_filename=dev_filename,
+                                                 embed_title=self.embed_title,
+                                                 num_hard_negatives=num_hard_negatives,
+                                                 num_negatives=num_negatives)
+
+        prediction_head = TextSimilarityHead(similarity_function=similarity_function)
+        bi_model = BiAdaptiveModel(
+            language_model1=self.query_encoder,
+            language_model2=self.passage_encoder,
+            prediction_heads=[prediction_head],
+            embeds_dropout_prob=0.1,
+            lm1_output_types=["per_sequence"],
+            lm2_output_types=["per_sequence"],
+            device=self.device,
+        )
+        bi_model.connect_heads_with_processor(self.processor.tasks, require_labels=True)
+
+        data_silo = DataSilo(processor=self.processor, batch_size=batch_size, distributed=False)
+
+        # 5. Create an optimizer
+        model, optimizer, lr_schedule = initialize_optimizer(
+            model=bi_model,
+            learning_rate=1e-5,
```

Comment:
make learning_rate, grad_acc_steps  args of `train()` with default values

Reply:
done